qiskit/dagcircuit/dagcircuit.py

```diff
@@ -289,14 +289,13 @@ def _add_op_node(self, op, qargs, cargs, condition=None):
         self._add_multi_graph_node(new_node)
         return new_node
 
-    def apply_operation_back(self, op, qargs=None, cargs=None, condition=None):
+    def apply_operation_back(self, op, qargs=None, cargs=None):
```

Comment:
We should deprecate the `condition` kwarg before removing it completely. In this case, if the user provides a `condition`, we can raise a warning, and set the condition on `op` to match.

Reply:
Thank you for reviewing the PR! Alright, will add a deprecation warning and make the release note for it. Thank you for the pointers :)